main.go

```diff
@@ -79,27 +89,30 @@ func main() {
 		klog.Fatal(err)
 	}
 
-	if err := machinev1.AddToScheme(mgr.GetScheme()); err != nil {
-		klog.Fatal("unable to add Machines to scheme")
-	}
-
-	directClient, err := client.New(mgr.GetConfig(), client.Options{
-		Scheme: mgr.GetScheme(),
-		Mapper: mgr.GetRESTMapper(),
-	})
-	if err != nil {
-		klog.Fatal("unable to set up client")
-	}
 	// Prevent the controller from caching node and machine objects.
 	// Stale nodes and machines can cause the approver to not approve certificates
 	// within a timely manner, leading to failed node bootstraps.
-	approverClient, err := client.NewDelegatingClient(client.NewDelegatingClientInput{
-		Client:      directClient,
+	uncachedManagementClient, err := client.NewDelegatingClient(client.NewDelegatingClientInput{
```

Comment:
I'm not sure why we woudn't just use the `mgr.GetAPIReader()` for this clients if we don't want to use the cache.

Reply:
IIRC we tried to use the API reader and found there was still an issue with it. I remember being deep into informer territory when debugging this and working out exactly what was happening, will see if I can find my notes.

This isn't the most elegant way to do this, we can actually pass uncached types into the management options, just need to get a PR up to clean that up. Will look at that later.

One advantage of excluding from the cache here is then you can't forget while coding to use the correct client, which I think is advantageous. If we blanket don't cache the object type, this isn't an issue when coding and makes the reconciler logic simpler.

Also, the client constructor in controller runtime can handle this via unstructured objects too so I don't think it's adding more complication on our side